src/ripple/ledger/ReadView.h

```diff
@@ -285,8 +310,28 @@ class DigestAwareReadView
     digest (key_type const& key) const = 0;
 };
 
-// For now
-using BasicView = ReadView;
+/** Run a functor on each SLE in a ReadView starting from the key start,
+    as long as the functor returns true.
+ */
+template <class Functor>
+void forEachSLE(ReadView const& view, Functor func, uint256 const& start = {})
```

Comment:
This is insanity, it visits every SLE! It should not be part of a public interface. If we need it somewhere, put the declaration and definition in the translation unit that uses it.